go/common/enclave.go

```diff
@@ -42,26 +43,26 @@ type Enclave interface {
 	SubmitL1Block(block L1Block, receipts L1Receipts, isLatest bool) (*BlockSubmissionResponse, error)
 
 	// SubmitTx - user transactions
-	SubmitTx(tx EncryptedTx) (EncryptedResponseSendRawTx, error)
+	SubmitTx(tx EncryptedTx) responses.RawTx
 
 	// SubmitBatch submits a batch received from the sequencer for processing.
 	SubmitBatch(batch *ExtBatch) error
 
 	// ExecuteOffChainTransaction - Execute a smart contract to retrieve data
 	// Todo - return the result with a block delay. To prevent frontrunning.
-	ExecuteOffChainTransaction(encryptedParams EncryptedParamsCall) (EncryptedResponseCall, error)
+	ExecuteOffChainTransaction(encryptedParams EncryptedParamsCall) responses.Call
 
 	// GetTransactionCount returns the nonce of the wallet with the given address (encrypted with the acc viewing key)
-	GetTransactionCount(encryptedParams EncryptedParamsGetTxCount) (EncryptedResponseGetTxCount, error)
+	GetTransactionCount(encryptedParams EncryptedParamsGetTxCount) responses.TxCount
 
 	// Stop gracefully stops the enclave
 	Stop() error
```

Comment:
should we create a type alias: SystemError, and return it in these places, to make it extra explicit?

Reply:
I had this for a bit, but it makes stuff confusing as the protocol right now cannot return user errors as such until it has a valid address (with a viewing key). Those user errors, like badly formed request params/inputs, end up being passed around as system errors which kind of defeated the purpose.